scripts/lib/CIME/XML/entry_id.py

```diff
@@ -360,7 +360,7 @@ def add_elements_by_group(self, srcobj, attributes=None, infile=None):
                 gname = "group_not_set"
             # If group with id=$gname does not exist in self.groups
             # then create the group node and add it to infile file
-            if gname not in self.groups.keys():
+            if gname not in list(self.groups.keys()):
```

Comment:
I don't think `list` is needed here, and I think you can just say `if gname not in self.groups:`